File: AkkaCustomExtension/src/main/scala/org/akka/essentials/custom/extension/example/MySQLJDBCSettings.scala
package org.akka.essentials.custom.extension.example
import akka.actor.ExtensionIdProvider
import akka.actor.ExtensionId
import akka.actor.ExtendedActorSystem

object MySQLJDBCSettings extends ExtensionId[MySQLJDBCSettingsImpl] with ExtensionIdProvider {

  override def lookup = MySQLJDBCSettings

  override def createExtension(system: ExtendedActorSystem) = new MySQLJDBCSettingsImpl(system.settings.config)
}

File: AkkaCustomExtension/src/main/java/org/akka/essentials/custom/extension/example/MySQLJDBCSettings.java
package org.akka.essentials.custom.extension.example;

import akka.actor.AbstractExtensionId;
import akka.actor.ExtendedActorSystem;
import akka.actor.ExtensionIdProvider;

public class MySQLJDBCSettings extends
		AbstractExtensionId<MySQLJDBCSettingsImpl> implements
		ExtensionIdProvider {

	public final static MySQLJDBCSettings SettingsProvider = new MySQLJDBCSettings();

	public MySQLJDBCSettings lookup() {
		return MySQLJDBCSettings.SettingsProvider;
	}

	public MySQLJDBCSettingsImpl createExtension(ExtendedActorSystem system) {
		return new MySQLJDBCSettingsImpl(system.settings().config());
	}
}